src/Umbraco.Cms.Integrations.Automation.Zapier/Components/ZapContentConfigComponent.cs

```diff
@@ -1,15 +1,67 @@
-﻿using Umbraco.Cms.Integrations.Automation.Zapier.Migrations;
+﻿
+using Umbraco.Cms.Integrations.Automation.Zapier.Migrations;
+
+#if NETCOREAPP
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Core.Events;
+using Umbraco.Cms.Core.Migrations;
+using Umbraco.Cms.Core.Notifications;
+using Umbraco.Cms.Core.Scoping;
+using Umbraco.Cms.Core.Services;
+using Umbraco.Cms.Infrastructure.Migrations;
+using Umbraco.Cms.Infrastructure.Migrations.Upgrade;
+#else
 using Umbraco.Core;
 using Umbraco.Core.Composing;
 using Umbraco.Core.Logging;
 using Umbraco.Core.Migrations;
 using Umbraco.Core.Migrations.Upgrade;
 using Umbraco.Core.Scoping;
 using Umbraco.Core.Services;
+#endif
 
 namespace Umbraco.Cms.Integrations.Automation.Zapier.Components
 {
+#if NETCOREAPP
+    public class ZapContentConfigMigration : INotificationHandler<UmbracoApplicationStartingNotification>
+    {
+        private readonly IMigrationPlanExecutor _migrationPlanExecutor;
+        
+        private readonly IScopeProvider _scopeProvider;
+        
+        private readonly IKeyValueService _keyValueService;
+        
+        private readonly IRuntimeState _runtimeState;
+
+        public ZapContentConfigMigration(
+            IScopeProvider scopeProvider,
+            IMigrationPlanExecutor migrationPlanExecutor,
+            IKeyValueService keyValueService,
+            IRuntimeState runtimeState)
+        {
+            _migrationPlanExecutor = migrationPlanExecutor;
+
+            _scopeProvider = scopeProvider;
+
+            _keyValueService = keyValueService;
+
+            _runtimeState = runtimeState;
+        }
+
+        public void Handle(UmbracoApplicationStartingNotification notification)
+        {
+            if(_runtimeState.Level < RuntimeLevel.Run) return;
```

Comment:
Same comment applies to this class with regard it maybe being better as two files.

You can also look to move the migration plan out into it's own class, which then needs a smaller set of compiler directive include/excludes (see [this example](https://github.com/umbraco/UI-Examples/blob/master/src/Our.Umbraco.UiExamples/Migrations/UiExamplesMigrationPlan.cs)).